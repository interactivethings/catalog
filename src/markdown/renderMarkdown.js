import marked from './marked-react';
import ReactRenderer from './ReactRenderer';

let MARKDOWN_CONFIG = {
  gfm: true,
  breaks: true,
  sanitize: false,
  smartLists: true,
  smartypants: true
};

export default ({text, renderer}) => {
  return marked(
    text,
    {
      ...MARKDOWN_CONFIG,
      renderer: Object.assign(new ReactRenderer(), renderer)
    }
  );
};
