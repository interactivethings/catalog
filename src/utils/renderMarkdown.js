import marked from './react-markdown';

let MARKDOWN_CONFIG = {
  gfm: true,
  breaks: true,
  sanitize: false,
  smartLists: true,
  smartypants: true
};

export default ({text, renderer}) => {
  return marked(
    text,
    {
      ...MARKDOWN_CONFIG,
      renderer: Object.assign(new marked.Renderer(), renderer)
    }
  );
};
