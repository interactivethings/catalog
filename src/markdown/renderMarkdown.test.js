import renderMarkdown from "./renderMarkdown";

const md = `
# Hello

World

- One
  - Two
- Three

A [link](http://www.interactivethings.com/) and some **bold** and *italic* text.

> Block quotes rock
`;

test("Renders some Markdown", () => {
  expect(
    renderMarkdown({
      text: md
    })
  ).toMatchSnapshot();
});

test("Renders some Markdown with custom renderers", () => {
  expect(
    renderMarkdown({
      text: md,
      renderer: {
        paragraph() {
          return "HELLO PARAGRAPH";
        },
        listitem() {
          return "HELLO LIST ITEM";
        }
      }
    })
  ).toMatchSnapshot();
});
