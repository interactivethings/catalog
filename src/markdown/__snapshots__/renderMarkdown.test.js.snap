// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders some Markdown 1`] = `
Array [
  <Heading
    level={1}
    slug="hello"
    text={
        Array [
            "Hello",
          ]
    }
/>,
  <p>
    World
</p>,
  <ul>
    <li>
        One
        <ul>
            <li>
                Two
            </li>
        </ul>
    </li>
    <li>
        Three
        
    </li>
</ul>,
  <p>
    A 
    <Link
        title={null}
        to="http://www.interactivethings.com/"
    >
        link
    </Link>
     and some 
    <strong>
        bold
    </strong>
     and 
    <em>
        italic
    </em>
     text.
</p>,
  <blockquote>
    <p>
        Block quotes rock
    </p>
</blockquote>,
]
`;

exports[`Renders some Markdown with custom renderers 1`] = `
Array [
  <Heading
    level={1}
    slug="hello"
    text={
        Array [
            "Hello",
          ]
    }
/>,
  "HELLO PARAGRAPH",
  <ul>
    HELLO LIST ITEM
    HELLO LIST ITEM
</ul>,
  "HELLO PARAGRAPH",
  <blockquote>
    HELLO PARAGRAPH
</blockquote>,
]
`;
